agenda/presentation/src/main/java/com/gkp/agenda/presentation/detail/AgendaItemDetailViewModel.kt

```diff
@@ -0,0 +1,65 @@
+package com.gkp.agenda.presentation.detail
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import androidx.navigation.toRoute
+import com.gkp.agenda.domain.AgendaRepository
+import com.gkp.agenda.domain.model.AgendaItem
+import com.gkp.agenda.presentation.detail.navigation.AgendaItemType
+import com.gkp.agenda.presentation.detail.navigation.AgendaItemDetailScreenRoute
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+class AgendaItemDetailViewModel(
+    savedStateHandle: SavedStateHandle,
+    private val agendaRepository: AgendaRepository,
+) : ViewModel() {
+
+    var uiState by mutableStateOf(AgendaItemDetailUiState())
+        private set
+
+    init {
+        val detailScreenRoute = savedStateHandle.toRoute<AgendaItemDetailScreenRoute>()
+        val itemId = detailScreenRoute.taskId
+        val itemType = detailScreenRoute.agendaItemType
+
+        if (itemId.isNotBlank()) {
```

Comment:
Passing in blank IDs to indicate they don't exist is wrong design. Non-nullable types communicate that they are mandatory and always exist. If you then bypass this by passing in empty invalid IDs, this is everything but obvious. Nullable types are used for exactly that: Indicating a value may or may not exist. If you encounter a nullable field somewhere in a codebase, you know there might be cases it isn't there (which you don't know if a field is non-nullable but can still not exist). 

Long story short: I'd recommend to make the ID nullable